### ISSUE TITLE
incorrect error CS1525 when CS1026 is expected

### ISSUE THREAD

**Issue by kasperk81** [dotnet/roslyn#64440]
**Version Used**: .net 7 rc2

**Steps to Reproduce**:

compile
```c#
public class C
{
    public void M()
    {
        var sum = S(5000,
            5060, // comment
            5061, // another comment
        );
    }

    public int S(int x, int y, int z) => x + y + z;
}
```

it doesn't like the trailing comma after `5061` but the error message is wrong.

**Expected Behavior**:

error: `/C.cs(7,18): error CS1026: ) expected`

**Actual Behavior**:

error: `/C.cs(7,18): error CS1525: Invalid expression term ')'`

**Comment by CyrusNajmabadi** [comment]
I'm not seeing a problem with these error message. There's no expression given between the last comma and the close paren.  So the parser is issuing an error saying that that missing expression is invalid.
